addons/crm/static/src/js/systray_activity_menu.js

```diff
@@ -22,5 +22,37 @@ ActivityMenu.include({
         }
         return this._super(...arguments);
     },
+
+    //-----------------------------------------
+    // Handlers
+    //-----------------------------------------
+
+    /**
+     * @private
+     * @override
+     */
+    _onActivityFilterClick: function (event) {
+        // fetch the data from the button otherwise fetch the ones from the parent (.o_mail_preview).
+        var data = _.extend({}, $(event.currentTarget).data(), $(event.target).data());
```

Comment:
Strange to mix the data and then hardcoding the context like that.

Couldn't we specify static data (define 2 ir.ui.view, one for "my" and another one), and call this.do_action directly, without additional context ?

@tde-banana-odoo What do you think ?

Reply:
Well actually it is already done like that in _onActivityFilterClick in mail; it builds a standard action and applies filters on it. You are right this is a bit odd.

It seems cleaner to have several actions defined. Drawback is that you have to define several actions (XX_overdue, XX_today, XX_upcoming) as you have, as far as I remember, 3 possible filters in systray.

We could have a mwat-mwat solution
  * one action for "my" (setting search_default_activities_overdue + search_default_activities_today in XML)
  * one action for "not my" (adding filters based on data.filter)

Do you think this would be cleaner ?

Some additional comments
  search_default_mymeetings -> there is no my meetings filter on lead ?
  default_mode -> why ?
  force_search_count -> it is really required ? Not sure what this is about ...

Reply:
Yep could be an interesting trade-off. 

Concerning force_search_count, there is a comment about that, but it seems hackish too.